### PR TITLE
Add support for Typha, the sync-offload daemon.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description
This PR allows Felix to use a remote Syncer connection via the [Typha](https://github.com/projectcalico/typha) fan-out daemon.  By offloading the work of the Syncer, we reduce the watcher burden on the datastore.

This support is currently experimental and the API is subject to breaking changes!.

## Todos
- [x] Integration tests (delete as appropriate) In plan
- [x] Documentation (in plan)

### CODE REVIEW

**Comment by neiljerram** [comment]
@fasaxc This fix seems to be needed to the coreos/etcd Git ID: https://github.com/neiljerram/felix/commit/27fd85e4fe1a4f6ef305d26a09d623b2db43dff0

File: felix.go [hunk 1]
```diff
@@ -778,3 +809,51 @@ func (fc *DataplaneConnector) Start() {
...
+	if err != nil {
+		log.WithError(err).Error("Unable to create Kubernetes config.")
+		return "", err
+	}
+	clientset, err := kubernetes.NewForConfig(k8sconf)
+	if err != nil {
+		log.WithError(err).Error("Unable to create Kubernetes client set.")
+		return "", err
+	}
+	svcClient := clientset.CoreV1().Services(configParams.TyphaK8sNamespace)
```
Comment (neiljerram) [inline felix.go]: Does namespace need to be non-"" ?  If so, we should check that above with TyphaK8sServiceName.

File: config/config_params.go [hunk 2]
```diff
@@ -103,6 +103,11 @@ type Config struct {
 	EtcdCaFile    string   `config:"file(must-exist);;local"`
 	EtcdEndpoints []string `config:"endpoint-list;;local"`
 
+	TyphaServiceEnvVarPfx string `config:"string;"`
```
Comment (neiljerram) [inline config/config_params.go]: Now unused?

File: felix.go (hunk 1 above)
Comment (fasaxc) [inline felix.go]: I've tackled that in the config; marking that config param as non-zero.

**Pull request merged**